src/scipp/plotting/formatters.py

```diff
@@ -50,13 +50,15 @@ class DateFormatter:
     Format datetime ticks: adjust the time precision and update
     offset according to the currently displayed range.
     """
-    def __init__(self, offset, dim):
+    def __init__(self, coord, offset, dim):
+        self.coord_values = coord.values
         self.offset = offset
         self.dim = dim
         self.indicators = []
 
     def formatter(self, val, pos, axis=None, get_axis_bounds=None, set_axis_label=None):
-        d = (self.offset + (int(val) * self.offset.unit)).value
+        index = np.abs(self.coord_values - val).argmin()
+        d = (self.offset + (index * self.offset.unit)).value
```

Comment:
So all you need is `index`, and not something like `self.label_values[index]`?

Reply:
Well, as far as I understand the existing code, for datetime the labels are not looked up but *computed*?

Reply:
i think it's fine.